pennylane/ops/qubit/arithmetic_ops.py

```diff
@@ -305,3 +310,257 @@ def compute_decomposition(wires):
+            the operation acts on.
+
+    **Example**
+
+    >>> dev = qml.device("default.qubit", wires=3)
+    >>> @qml.qnode(dev)
+    ... def circuit(state, value, geq):
+    ...     qml.BasisState(np.array(state), wires=range(3))
+    ...     qml.IntegerComparator(value, geq=geq, wires=range(3))
+    ...     return qml.state()
+    >>> circuit([1, 0, 1], 1, True).reshape(2, 2, 2)[1, 0, 0]
+    (1+0j)
+    >>> circuit([0, 1, 0], 3, False).reshape(2, 2, 2)[0, 1, 1]
+    (1+0j)
+    """
+    is_self_inverse = True
+    num_wires = AnyWires
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    grad_method = None
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        value,
+        geq=True,
+        wires=None,
+        do_queue=True,
+    ):
+
+        if not isinstance(value, int):
+            raise ValueError(f"The compared value must be an int. Got {type(value)}.")
+        if wires is None:
+            raise ValueError("Must specify wires that the operation acts on.")
+        if len(wires) > 1:
+            control_wires = Wires(wires[:-1])
+            wires = Wires(wires[-1])
+        else:
+            raise ValueError(
+                "IntegerComparator: wrong number of wires. "
+                f"{len(wires)} wire(s) given. Need at least 2."
+            )
+
+        total_wires = control_wires + wires
+
+        self.hyperparameters["control_wires"] = control_wires
+        self.hyperparameters["target_wires"] = wires
+        self.hyperparameters["value"] = value
+        self.hyperparameters["geq"] = geq
+        self.geq = geq
+        self.value = value
+
+        super().__init__(wires=total_wires, do_queue=do_queue)
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        return base_label or f">={self.value}" if self.geq else f"<{self.value}"
+
+    # pylint: disable=unused-argument
+    @staticmethod
+    def compute_matrix(
+        value=None, control_wires=None, geq=True, **kwargs
+    ):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.IntegerComparator.matrix`
+
+        Args:
+            value (int): The value :math:`L` that the state's decimal representation is compared against.
+            control_wires (Union[Wires, Sequence[int], or int]): wires to place controls on
+            geq (bool): If set to `True`, the comparison made will be :math:`n \geq L`. If `False`, the comparison
+                made will be :math:`n < L`.
+
+        Returns:
+           tensor_like: matrix representation
+
+        **Example**
+
+        >>> print(qml.IntegerComparator.compute_matrix(2, [0,1]))
```

Comment:
(Formatting would be adjusted with `black`.)
```suggestion
        >>> print(qml.IntegerComparator.compute_matrix(2, [0, 1]))
```